fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonExternalTable.java

```diff
@@ -63,18 +66,22 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.OptionalLong;
 import java.util.Set;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 import java.util.stream.Collectors;
 
-public class PaimonExternalTable extends ExternalTable implements MTMVRelatedTableIf, MTMVBaseTableIf {
+public class PaimonExternalTable extends ExternalTable implements MTMVRelatedTableIf, MTMVBaseTableIf, MvccTable {
 
     private static final Logger LOG = LogManager.getLogger(PaimonExternalTable.class);
 
+    private final ReentrantReadWriteLock snapshotRwLock = new ReentrantReadWriteLock(true);
```

Comment:
could we avoid use lock at all? fair write read lock will lead to dead lock finally